### ISSUE TITLE
Error in parsing IA32 pextrw instruction

### ISSUE THREAD

**Issue by LRGH** [bdcht/amoco#28]
Here is the current output:

``` python
>>> from amoco.arch.x86 import cpu_x86
>>> i = cpu_x86.disassemble('\x66\x0f\xc5\xc1\x00')
>>> str(i)
'pextrw      xmm0, ecx, 0x0'
```

While it should be `pextrw eax, xmm1, 0`

**Comment by LRGH** [comment]
Proposed patch:

``` python
--- a/amoco/arch/x86/spec_sse.py
+++ b/amoco/arch/x86/spec_sse.py
@@ -239,24 +239,10 @@ def sse_pd(obj,Mod,REG,RM,data):
     obj.bytes += pack(imm)
     obj.type = type_data_processing

-# mmx, r32/m16, imm8
-@ispec_ia32("*>[ {0f}{c4} /r ]", mnemonic="PINSRW")
-def sse_pd(obj,Mod,REG,RM,data):
-    if not check_nopfx(obj,set_opdsz_32): raise InstructionError(obj)
-    op2,data = getModRM(obj,Mod,RM,data)
-    if op2._is_mem: op2.size = 16
-    op1 = env.getreg(REG,64)
-    obj.operands = [op1,op2]
-    if data.size<8: raise InstructionError(obj)
-    imm = data[0:8]
-    obj.operands.append(env.cst(imm.int(),8))
-    obj.bytes += pack(imm)
-    obj.type = type_data_processing
-
 # r32, mmx, imm8
 @ispec_ia32("*>[ {0f}{c5} /r ]", mnemonic="PEXTRW")
 def sse_pd(obj,Mod,REG,RM,data):
-    if not check_nopfx(obj,set_opdsz_64): raise InstructionError(obj)
+    if not check_66(obj,set_opdsz_128): raise InstructionError(obj)
     op2,data = getModRM(obj,Mod,RM,data)
     if op2._is_mem: raise InstructionError(obj)
     op1 = env.getreg(REG,32)
@@ -761,7 +747,6 @@ def sse_pd(obj,Mod,REG,RM,data,_inv):
 # r/m32, xmm, imm8
 @ispec_ia32("*>[ {0f}{3a}{14} /r ]", mnemonic="PEXTRB", _inv=True)
 @ispec_ia32("*>[ {0f}{c4}     /r ]", mnemonic="PINSRW", _inv=False)
-@ispec_ia32("*>[ {0f}{c5}     /r ]", mnemonic="PEXTRW", _inv=False)
 @ispec_ia32("*>[ {0f}{3a}{15} /r ]", mnemonic="PEXTRW", _inv=True)
 @ispec_ia32("*>[ {0f}{3a}{16} /r ]", mnemonic="PEXTRD", _inv=True)
 @ispec_ia32("*>[ {0f}{3a}{17} /r ]", mnemonic="EXTRACTPS", _inv=True)
```

**Comment by bdcht** [comment]
Hi, thanks for reporting this, should be fixed by commit 2dd1924.

**Comment by LRGH** [comment]
Thanks.
It is fixed by your commit, but you did not delete the old (and invalid) definitions of PINSRW and PEXTRW (lines 242 to 268), which are overwritten later in the file.
